### DISCUSSION TITLE
[HELP] Get a child widget from a Notebook page

### DISCUSSION THREAD

**Discussion by Wanpaku** [gtk-rs/gtk4-rs#1700] [Q&A] [answered]
I have a Notebook with Listbox as child for every page.
When I try to get a child for current page I get only base class Widget.
This Widget has GtkListBox name but it hasn't any functionality of ListBox.

`let current_page_index = my_notebook.current_page();
 let current_page_child =
                        my_notebook.nth_page(current_page_index).unwrap();`

It hasn't append() function which I need, for example.
How can I get fully functional ListBox child from Notebook current page?

**Comment by bilelmoussaoui** [comment, marked as answer]
You have to cast the gtk::Widget as gtk::ListBox. `let listbox = current_page_child.downcast::<gtk::ListBox>().unwrap()`

**Comment by bilelmoussaoui** [comment]
Also please use discussions for any questions you might have

**Reply by Wanpaku** [reply]
OK. Thank you.
